src/env/env_utils.c

```diff
@@ -0,0 +1,72 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   env_utils.c                                        :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: alopez-g <marvin@42.fr>                    +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2022/10/15 21:14:27 by alopez-g          #+#    #+#             */
+/*   Updated: 2022/10/15 22:14:30 by alopez-g         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "env.h"
+
+char	*get_key_value_pair(char *key, char *value)
+{
+	char	*pair;
+	char	*tmp;
+
+	tmp = ft_strjoin(key, "=");
+	pair = ft_strjoin(tmp, value);
+	free(tmp);
+	return (pair);
+}
+
+t_error_code	new_var(char ***env, char *key, char *value)
+{
+	if (!*env)
+	{
+		*env = (char **)malloc(sizeof(char *) + 1);
+		*(*env + 1) = NULL;
```

Comment:
how do you know there is mem in *(*env + 1)
you only checked if (!*env)

Reply:
I cann access *(*env + 1) because I just malloced space for it, but  have changed it to char **)malloc(sizeof(char *)), without + 1, it was not needed, it just need to initalize *env if it is not